### ISSUE TITLE
Unable to identify ArgParser for...

### ISSUE THREAD

**Issue by dethpickle** [Chris7/djangui#11]
I was having troubles trying to import any of my own argparse scripts and fell back to [PSF's tutorial](https://docs.python.org/2/howto/argparse.html).  I modified it slightly so it'd have a main module:

``` python
import argparse

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description="calculate X to the power of Y")
    group = parser.add_mutually_exclusive_group()
    group.add_argument("-v", "--verbose", action="store_true")
    group.add_argument("-q", "--quiet", action="store_true")
    parser.add_argument("x", type=int, help="the base")
    parser.add_argument("y", type=int, help="the exponent")
    args = parser.parse_args()
    answer = args.x**args.y

    if args.quiet:
        print answer
    elif args.verbose:
        print "{} to the power {} equals {}".format(args.x, args.y, answer)
    else:
        print "{}^{} == {}".format(args.x, args.y, answer)
```

When I run that I get the same error I get with my own scripts:

```
python manage.py addscript /home/dethpickle/tsutilweb/checkmate/exponent.py
/home/dethpickle/.virtualenvs/tsutilweb/local/lib/python2.7/site-packages/request/managers.py:95: RemovedInDjango18Warning: `RequestManager.get_query_set` method should be renamed `get_queryset`.
  class RequestManager(models.Manager):

Converting /home/dethpickle/tsutilweb/checkmate/exponent.py
Unable to identify ArgParser for /home/dethpickle/tsutilweb/djangui_scripts/exponent_djKAfPl.py:
Converted 0 scripts
```

I'm on Python 2.7.6.  I was trying to use the PSF tutorial as a way of baselining that I wasn't personally attempting anything non-standard.  Is there a chance you could post one of your sample scripts from your heroku site so I could apples to apples this?

**Comment by streetmapp** [comment]
I was having issues as well when getting setup, but was able to get this simple script to work once I was set up. Was using Python 2.7.9 and Django 1.8.2. So not sure if there'd be any other major differences, but hopefully this helps. 

https://gist.github.com/streetmapp/dc0ae843529adb912749

—

**Comment by Chris7** [comment]
Sure, I'll add sample scripts to the docs. There are some rules you need to follow, so it makes sense.

**Comment by Chris7** [comment]
A quick example is here:
https://github.com/Chris7/django-djangui/blob/master/README.md#script-guide

You can also see the test scripts here:
https://github.com/Chris7/django-djangui/tree/master/djangui/tests/scripts

**Comment by dethpickle** [comment]
Fabulous.  That got me over that hump.

I suppose I was declaring all the argparse in main because I figured if anyone was ever going to invoke it, it must be coming through main.  That's not a good reason... just a reason.  Thanks for all the examples.  I've got my simple and full-blown items converting now.

I have scripts loaded and everything.  It looks as though it should be working but my ^djangui/ pages are coming up blank.  I actually think that's a good sign.  I can see through the debug toolbar that I'm loading `djangui/templates/djangui_home.html` and I have scripts in the `djangui_scripts` dictionary.

This issue is closed.  Let me mull over what what else is going on.  I half think I have a template that squashing yours somehow.  Thanks for your help!
